internal/app/machined/pkg/controllers/network/hostname_config.go

```diff
@@ -97,7 +105,36 @@ func (ctrl *HostnameConfigController) Run(ctx context.Context, r controller.Runt
 			defaultAddr = addrs.(*network.NodeAddress) //nolint:errcheck,forcetypeassert
 		}
 
-		specs = append(specs, ctrl.getDefault(defaultAddr))
+		if cfgProvider != nil && cfgProvider.Machine().Features().StableHostnameEnabled() {
```

Comment:
we should wait for the `cfgProvider != nil` before we make a decision on which hostname to use, otherwise Talos will always first do `talos-192-168-1-1` and then replace it with stable machine id

Reply:
Wait like `cfgProvider != nil { continue }`? Or just get `ctrl.getDefault(defaultAddr)` under the `if`? Because former changes the current behavior of the controller.

Reply:
don't do any default hostname until we get machine config... this will allow us to avoid these endless hostname changes

Reply:
(not `continue`, just skip the default hostname until cfgProvider is around)

Reply:
done